### PR TITLE
Use beacon state interfaces across

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

Using interfaces defined in #8544 for beacon state arguments. This is to get us started with HF1

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

N/A

### CODE REVIEW

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
Blocking until 3 approves

File: beacon-chain/core/epoch/precompute/new.go
Comment (nisdas) [inline beacon-chain/core/epoch/precompute/new.go:22]: same as what we discussed earlier, the interface cannot be nil. So the check will not check for anything.

**Review by nisdas — COMMENTED** [review]
Main thing that we would need to look out for is nil check on the state. (Whereas the check for a pointer would be positive the same isn't the case for an interface.

File: beacon-chain/powchain/service_test.go
Comment (rkapka) [inline beacon-chain/powchain/service_test.go:437]: This change seems unrelated to the PR. Does the original test fail after the refactor?

File: beacon-chain/state/getters.go
Comment (rkapka) [inline beacon-chain/state/getters.go:573]: This change should be in its own PR.

File: beacon-chain/sync/pending_attestations_queue_test.go
Comment (rkapka) [inline beacon-chain/sync/pending_attestations_queue_test.go:140]: This change should be in another PR.

**Review by rkapka — COMMENTED** [review]
I would like to see a few things extracted out to a separate PR. Because the diff here is large, I would avoid non-essential refactoring.

**Comment by terencechain** [comment]
@nisdas @rkapka 

Feedbacks addressed. I removed all the nil checks and fixed the tests. The issue was we had tests that depends solely on `hasInnerState` to proceed, where b is defined as `nil` in tests.
```go
func (b *BeaconState) hasInnerState() bool {
	return b != nil && b.state != nil
}
```  

Now the tests had to be adapted to `(*stateTrie.BeaconState)(nil)` instead

File: beacon-chain/powchain/service_test.go
Comment (terencechain) [reply beacon-chain/powchain/service_test.go:437]: Rationale: https://github.com/prysmaticlabs/prysm/pull/8552#issuecomment-791529031

**Review by rkapka — APPROVED** [review]

**Comment by rkapka** [comment]
Why is this PR labeled as blocked?

**Review by rkapka — DISMISSED** [review]

**Comment by terencechain** [comment]
> Why is this PR labeled as blocked?

https://github.com/prysmaticlabs/prysm/pull/8552#issuecomment-790929064

**Review by rauljordan — DISMISSED** [review]

**Comment by terencechain** [comment]
Removing blocked. This PR has gotten 4 approvals so far. Unfortunately conflict is adding up and we could be reviewing the commit diffs since last approval

**Review by rkapka — APPROVED** [review]

**Review by rauljordan — APPROVED** [review]

**Pull request merged**
